tests/test_intl/test_intl.py

```diff
@@ -681,50 +682,188 @@ def test_translation_progress_classes_true(app):
     assert len(doctree[0]) == 20
 
 
+class _MockClock:
+    """Object for mocking :func:`time.time_ns` (if needed).
+
+    Use :meth:`sleep` to make this specific clock sleep for some time.
+    """
+
+    def time(self) -> int:
+        """Nanosecond since 'fake' epoch."""
+        raise NotImplementedError
+
+    def sleep(self, ds: float) -> None:
+        """Sleep *ds* seconds."""
+        raise NotImplementedError
+
+
+class _MockWindowsClock(_MockClock):
+    """Object for mocking :func:`time.time_ns` on Windows platforms.
+
+    The result is in 'nanoseconds' but with a microsecond resolution
+    so that the division by 1_000 does not cause rounding issues.
+    """
+
+    def __init__(self) -> None:
+        self.us: int = 0  # current microsecond 'tick'
+
+    def time(self) -> int:
+        ret = 1_000 * self.us
+        self.us += 1
+        return ret
+
+    def sleep(self, ds: float) -> None:
+        self.us += int(ds * 1e6)
+
+
+class _MockUnixClock(_MockClock):
+    """Object for mocking :func:`time.time_ns` on Unix platforms.
+
+    Since nothing is needed for Unix platforms, this object acts as
+    a proxy so that the API is the same as :class:`_MockWindowsClock`.
+    """
+
+    def time(self) -> int:
+        return time.time_ns()
+
+    def sleep(self, ds: float) -> None:
+        time.sleep(ds)
+
+
+@pytest.fixture()
+def mock_time_and_i18n(
+    monkeypatch: pytest.MonkeyPatch,
+) -> tuple[pytest.MonkeyPatch, _MockClock]:
+    from sphinx.util.i18n import CatalogInfo
+
+    # save the 'original' definition
+    catalog_write_mo = CatalogInfo.write_mo
+
+    def mock_write_mo(self, locale, use_fuzzy=False):
+        catalog_write_mo(self, locale, use_fuzzy)
+        # ensure that the .mo file being written has a correct fake timestamp
+        _set_mtime_ns(self.mo_path, time.time_ns())
+
+    # see: https://github.com/pytest-dev/pytest/issues/363
+    with pytest.MonkeyPatch.context() as mock:
+        if os.name == 'posix':
```

Comment:
Is this preferable to checking `sys.platform`?

Reply:
That's how we skip tests so I think it's fine.